dash_agent/README.md

```diff
@@ -1,39 +1,174 @@
-<!--
-This README describes the package. If you publish this package to pub.dev,
-this README's contents appear on the landing page for your package.
+# Dash Agent 
 
-For information about how to write a good package README, see the guide for
-[writing package pages](https://dart.dev/guides/libraries/writing-package-pages).
+Dash agent is a framework enabling you to create and publish agents in CommandDash marketplace. 
 
-For general information about developing packages, see the Dart guide for
-[creating packages](https://dart.dev/guides/libraries/create-library-packages)
-and the Flutter guide for
-[developing packages and plugins](https://flutter.dev/developing-packages).
--->
-
-TODO: Put a short description of the package here that helps potential users
-know whether this package might be useful for them.
+## Getting started
 
-## Features
+You don't need to manually install the package. As the package comes pre-installed with the agent project that you can create using [dash-cli](). If it's not already activated. You can do so by running the following command in the terminal:
 
-TODO: List what your package can do. Maybe include images, gifs, or videos.
+```shell
+dart pub global activate dash-cli
+```
 
-## Getting started
+Now, run the below shared command in the terminal once again:
 
-TODO: List prerequisites and provide or point to information on how to
-start using the package.
+```shell
+dash-cli create {{agent}}
+```
 
 ## Usage
 
-TODO: Include short and useful examples for package users. Add longer examples
-to `/example` folder.
+This package contain the building blocks for creating dash agents:
+
+### AgentConfiguration
+
+The main part of the framework that glues together your agent configuration is `AgentConfiguration`. It is the base class for creating agent.
+
+Sample example of `AgentConfiguration` for demonstration purpose:
+
+```dart
+class MyAgent extends AgentConfiguration {
+  final docsSource = DocsDataSource();
+
+  @override
+  List<DataSource> get registeredDataSources => [docsSource];
+
+  @override
+  List<Command> get registerSupportedCommands =>
+      [AskCommand(docsSource: docsSource)];
+}
+```
+
+The above sample AgentConfiguration object taking data sources and supported commands. Both of them are explained below:
+
+- `DataSource`: Data sources will enable you to pass on any form of data that your agent might need to perform its intended tasks.
+- `Command`: Commands are the specialised tasks you want your agents to perform (like understanding the developer query, refactoring, code generation, etc).
```

Comment:
```suggestion
The above AgentConfiguration registers your data sources and supported commands.

- `DataSource`: Data sources enables you to provide any form of data that your agent might need to perform its intended tasks.
- `Command`: Commands are the specialised tasks you want your agents to perform (like answering developer's query, refactoring, code generation, etc).
```

Reply:
Foolish leaving words at time mistake of me!🤦‍♂️